syntax = "proto2";

package tagset.dep_labels;

// Dependency Labels.
enum Tag {
  UNKNOWN_LABEL = 0;
  acl = 1;              // adjective clause
  advcl = 2;            // adverbial clause
  advmod = 3;           // adverbial modifier
  advmod_emph = 4;      // adverbial modifier ?? 
  amod = 5;             // adjective modifier
  appos = 6;            // apposition
  aux = 7;              // auxiliary
  aux_q = 8;            // question auxiliary
  case = 9;             // case
  cc = 10;              // coordinating conjunction
  cc_preconj = 11;      // cc preconj
  ccomp = 12;           // complementizer clause
  clf = 13;             // classifier
  compound = 14;        // compound
  compound_lvc = 15;    // light verb compound
  compound_redup = 16;  // reduplicated compound
  conj = 17;            // conjunction
  cop = 18;             // copula
  csubj = 19;           // clausal subject
  dep = 20;             // dependent
  det = 21;             // determiner
  discourse = 22;       // discourse marker
  dislocated = 23;      // dislocated element
  fixed = 24;           // fixed
  flat = 25;            // flat
  goeswith = 26;        // goes with
  iobj = 27;            // indirect object
  list = 28;            // list of items
  mark = 29;            // complementizer
  nmod = 30;            // nominal modifier
  nmod_poss = 31;       // nominal modifier: ??
  nsubj = 32;           // nominal subject
  nummod = 33;          // numerical modification
  obj = 34;             // direct object
  obl = 35;             // oblique object
  orphan = 36;           // orpham
  parataxis = 37;       // parataxis
  punct = 38;           // punctuation
  root = 39;            // root
  vocative = 40;        // vocative
  xcomp = 41;           // nonfinite clausal complement
  TOP = 42;             // dummy dep label for the top token.
}

// fine_to_coarse_map_by_index
//  0: 0
//  # adjuncts
//  2: 1, 7: 1, 8: 1, 18: 1, 35: 1, 40: 1,
//  # arguments
//  12: 2, 19: 2, 27: 2, 32: 2, 34: 2, 41: 2,
//  # modifiers
//  1: 3, 3: 3, 4: 3, 5: 3, 6: 3, 9: 3, 13: 3, 14: 3, 15: 3, 16: 3, 20:3, 21:3,
//  22:3, 23:3, 24:3, 25:3, 26:3, 28:3, 29:3, 30:3, 31:3, 33:3, 36:3, 38:3,
//  # root
//  39: 4,
//  # cc
//  10: 5, 11: 5,
//  # conjunct
//  17: 6, 37: 6,
//  # TOP token
//  42: 7,

enum CoarseDepTag {
  UNKNOWN_COARSE = 0;
  adjunct = 1;
  argument = 2;
  modifier = 3;
  root_coarse = 4;
  cc_coarse = 5;
  conjunct_coarse = 6;
  TOP_COARSE = 7;
}
